cmake/TestImpactFramework/LYTestImpactFramework.cmake

```diff
@@ -439,6 +452,31 @@ function(ly_test_impact_write_config_file CONFIG_TEMPLATE_FILE BIN_DIR)
     message(DEBUG "Test impact framework post steps complete")
 endfunction()
 
+#! ly_test_impact_write_pytest_file: writes out the test information utilised by our TIAF testing tools, using the data derived from the build generation process.
+# 
+function(ly_test_impact_write_pytest_file)
+
+    # For each configuration type, compile the build info we need and add it to our array
+    set(build_configs "")
+    foreach(config_type ${LY_CONFIGURATION_TYPES})
+        set(config "${LY_TEST_IMPACT_WORKING_DIR}/${config_type}/${LY_TEST_IMPACT_PERSISTENT_DIR}/${LY_TEST_IMPACT_CONFIG_FILE_NAME}")
+        ly_file_read("cmake/TestImpactFramework/LyTestImpactBuildConfigEntry.in" build_config)
+        string(CONFIGURE ${build_config} build_config)
+        list(APPEND build_configs "${build_config}")
+    endforeach()
+ 
+    # Configure our list of entries
+    string(REPLACE ";" ",\n" build_configs "${build_configs}")
+    
+    # Configure and write out our test data file
+    ly_file_read("cmake/TestImpactFramework/LYTestImpactTestData.in" test_file)
```

Comment:
I wouldn't hard code the input file here, I would pass it in as a func arg as per the other functions writing out templates (e.g. see `ly_test_impact_write_config_file`).

Reply:
Done